src/main/scala/com/gu/fastly/Lambda.scala

```diff
@@ -35,20 +35,46 @@ class Lambda {
+          val jsonPurge = sendFastlyPurgeRequestForAjaxFile(event.payloadId, contentType)
+          val mapiCanonicalPurge = sendFastlyPurgeRequest(event.payloadId, Soft, config.fastlyMapiServiceId, makeMapiSurrogateKey(event.payloadId), config.fastlyMapiApiKey, contentType)
+          val aliasPurges = extractAliasPaths(event).fold(Seq.empty[Boolean])({
+            _.map { aliasPath =>
+              val dotcomAliasPurge = sendFastlyPurgeRequest(aliasPath, Soft, config.fastlyDotcomServiceId, makeDotcomSurrogateKey(aliasPath), config.fastlyDotcomApiKey, contentType)
+              val jsonAliasPurge = sendFastlyPurgeRequestForAjaxFile(aliasPath, contentType)
+              val mapiAliasPurge = sendFastlyPurgeRequest(aliasPath, Soft, config.fastlyMapiServiceId, makeMapiSurrogateKey(aliasPath), config.fastlyMapiApiKey, contentType)
+              dotcomAliasPurge && jsonAliasPurge && mapiAliasPurge
+            }
+          })
+          dotcomCanonicalPurge && jsonPurge && mapiCanonicalPurge && !aliasPurges.contains(false)
 
         case (ItemType.Content, EventType.RetrievableUpdate) =>
           val contentType = extractUpdateContentType(event)
-          sendFastlyPurgeRequest(event.payloadId, Soft, config.fastlyDotcomServiceId, makeDotcomSurrogateKey(event.payloadId), config.fastlyDotcomApiKey, contentType)
-          sendFastlyPurgeRequestForAjaxFile(event.payloadId, contentType)
-          sendFastlyPurgeRequest(event.payloadId, Soft, config.fastlyMapiServiceId, makeMapiSurrogateKey(event.payloadId), config.fastlyMapiApiKey, contentType)
+          val dotcomCanonicalPurge = sendFastlyPurgeRequest(event.payloadId, Soft, config.fastlyDotcomServiceId, makeDotcomSurrogateKey(event.payloadId), config.fastlyDotcomApiKey, contentType)
+          val jsonCanonicalPurge = sendFastlyPurgeRequestForAjaxFile(event.payloadId, contentType)
+          val mapiCanonicalPurge = sendFastlyPurgeRequest(event.payloadId, Soft, config.fastlyMapiServiceId, makeMapiSurrogateKey(event.payloadId), config.fastlyMapiApiKey, contentType)
+          val aliasPurges = extractAliasPaths(event).fold(Seq.empty[Boolean])({
+            _.map { aliasPath =>
+              val dotcomAliasPurge = sendFastlyPurgeRequest(aliasPath, Soft, config.fastlyDotcomServiceId, makeDotcomSurrogateKey(aliasPath), config.fastlyDotcomApiKey, contentType)
+              val jsonAliasPurge = sendFastlyPurgeRequestForAjaxFile(aliasPath, contentType)
+              val mapiAliasPurge = sendFastlyPurgeRequest(aliasPath, Soft, config.fastlyMapiServiceId, makeMapiSurrogateKey(aliasPath), config.fastlyMapiApiKey, contentType)
+              dotcomAliasPurge && jsonAliasPurge && mapiAliasPurge
+            }
+          })
+          dotcomCanonicalPurge && jsonCanonicalPurge && mapiCanonicalPurge && !aliasPurges.contains(false)
```

Comment:
This naming is useful

Reply:
aliasPurges.contains(false) has my attention... not obvious what that means... parsing.

Reply:
Does dotcomCanonicalPurge && jsonCanonicalPurge && mapiCanonicalPurge && !aliasPurges.contains(false)

Actually mean:
!(dotcomCanonicalPurge && jsonCanonicalPurge && mapiCanonicalPurge && aliasPurges).contains(false) ?

Reply:
The Seq[String] that is `aliasPaths` gets turned into a `Seq[Boolean]` when the requests are sent, thus any could be false so I'm checking that 100% were true and there were no failures. I wonder if there's a cleaner way to express it 🤔

Reply:
Got it ... Id failed to see that the 1st 3 args are Boolean not Seq[Boolean].... this expression is correct.